samples/net/zperf/src/main.c

```diff
@@ -29,4 +69,15 @@ void main(void)
 #ifdef CONFIG_NET_LOOPBACK_SIMULATE_PACKET_DROP
 	loopback_set_packet_drop_ratio(1);
 #endif
+
+#ifdef CONFIG_NET_DHCPV4
+	struct net_if *iface;
+
+	LOG_INF("Run dhcpv4 client");
+	net_mgmt_init_event_callback(&mgmt_cb, handler, NET_EVENT_IPV4_ADDR_ADD);
+	net_mgmt_add_event_callback(&mgmt_cb);
+
+	iface = net_if_get_default();
+	net_dhcpv4_start(iface);
```

Comment:
Generally this commit should not be needed. The sample uses NET_CONFIG lib, which triggers DHCPv4, if enabled. I had no problem obtaining an IPv4 address over DHCPv4 with other boards.

Reply:
Thank you for reminding me, I'll try it and remove this commit.